postgres_exporter/common/queries_nodemx.yml

```
@@ -101,8 +111,19 @@ ccp_nodemx_cpu:
         description: "CPU limit value in milli cores"
 
 ccp_nodemx_cpucfs:
-  query: "SELECT  monitor.cgroup_scalar_bigint('cpu.cfs_period_us') as period_us, 
-                  case when monitor.cgroup_scalar_bigint('cpu.cfs_quota_us') < 0 then 0 else monitor.cgroup_scalar_bigint('cpu.cfs_quota_us') end as quota_us"
+  query: "SELECT
+  CASE
+  WHEN monitor.cgroup_mode() = 'legacy' THEN
+    monitor.cgroup_scalar_bigint('cpu.cfs_period_us')
+  ELSE
+    (SELECT cgroup_array_bigint[2] FROM monitor.cgroup_array_bigint('cpu.max'))
+  END AS period_us, 
+  CASE
+  WHEN monitor.cgroup_mode() = 'legacy' THEN
+    (CASE WHEN monitor.cgroup_scalar_bigint('cpu.cfs_quota_us') < 0 THEN 0 ELSE monitor.cgroup_scalar_bigint('cpu.cfs_quota_us') END)
+  ELSE
+    (CASE WHEN (SELECT cgroup_array_bigint[1] FROM monitor.cgroup_array_bigint('cpu.max')) < 0 THEN 0 ELSE (SELECT cgroup_array_bigint[1] FROM monitor.cgroup_array_bigint('cpu.max')) END)
```

🤔 These can be shorter with [GREATEST](https://www.postgresql.org/docs/current/functions-conditional.html#FUNCTIONS-GREATEST-LEAST), I think. (The latter is only correct if SELECT can be eliminated.)

```suggestion
    GREATEST(monitor.cgroup_scalar_bigint('cpu.cfs_quota_us'), 0)
  ELSE
    GREATEST(monitor.cgroup_array_bigint('cpu.max')[1], 0)
```

---

Nice! I'll use GREATEST.